CardGame.playground/Contents.swift

```diff
@@ -78,21 +158,62 @@ import Foundation
 //: Create a protocol for a `CardGame`. It should have two requirements:
 //: * a gettable `deck` property
 //: * a `play()` method
-
+protocol CardGame {
+    var deck: Deck { get }
+    func play()
+}
 
 
 
 //: ## Step 13
 //: Create a protocol for tracking a card game as a delegate called `CardGameDelegate`. It should have two functional requirements:
 //: * a function called `gameDidStart` that takes a `CardGame` as an argument
 //: * a function with the following signature: `game(player1DidDraw card1: Card, player2DidDraw card2: Card)`
-
+protocol CardGameDelegate {
+    func gameDidStart(game: CardGame)
+    func game(player1DidDraw card1: Card, player2DidDraw card2: Card)
+}
 
 
 
 //: ## Step 14
 //: Create a class called `HighLow` that conforms to the `CardGame` protocol. It should have an initialized `Deck` as a property, as well as an optional delegate property of type `CardGameDelegate`.
-
+class HighLow: CardGame {
+    var deck: Deck
+    
+    var delegate: CardGameDelegate?
+    
+    init() {
+        self.deck = Deck.init()
+    }
+    
+    //draw 2 cards from the deck, one for player 1 and one for player 2.
+    //These cards will then be compared to see which one is higher.
+    //The winning player will be printed along with a description of the winning card.
+    func play() {
+        self.delegate?.gameDidStart(game: self)
+        
+        let player1Card = deck.drawCard()
+        let player2Card = deck.drawCard()
+        
+        self.delegate?.game(player1DidDraw: player1Card, player2DidDraw: player2Card)
+        
+        if player1Card < player2Card {
+            print("Player 2 wins with \(player2Card).")
+        } else if player1Card > player2Card {
+            print("Player 1 wins with \(player1Card).")
+        } else if player1Card == player2Card {
+            print("Round ends in a tie with \(player1Card).")
+        } else {
+            print("Rounds ends in a tie.")
```

Comment:
This last `else` is dead code, as the first three `if` statements cover every possible scenario. I would consider shortening to this:

``` swift
if player1Card < player2Card {
    print("Player 2 wins with \(player2Card).")
} else if player1Card > player2Card {
    print("Player 1 wins with \(player1Card).")
} else {
    print("Round ends in a tie with \(player1Card).")
}
```